src/Discord/Parts/Embed/Embed.php

```diff
@@ -469,6 +504,26 @@ public function setURL(string $url): self
         return $this;
     }
 
+    /**
+     * Checks to see if an url is valid to be used in embeds.
+     *
+     * @param ?string $url
+     *
+     * @return bool
+     */
+    protected function checkEmbedUrl(?string $url, array $allowed = ['http', 'https', 'attachment']): bool
+    {
+        if (null === $url) return true;
```

Comment:
`null` isn't using a valid url scheme. Instead of returning `true` to allow it to pass through the check, it should not be going into this check at all.

Instead of `if (! $this->checkEmbedUrl($url))` in other functions you should use `if (is_null($url) || ! $this->checkEmbedUrl($url))`